### DISCUSSION TITLE
EditExec - edit a file on the watch and exec it as Python

### DISCUSSION THREAD

**Discussion by drtonyr** [wasp-os/wasp-os#450] [Ideas]
Back in the 80's I wanted a watch that ran Lisp, it would just be cool to be able to write code on your own watch.  40 years later and we can do better, I'd much prefer editing and running Python on my watch than Lisp.  EditExec is a very simple text editor that can exec() the result.   Here is 'hello world':

![EditExecApp](https://github.com/wasp-os/wasp-os/assets/30016701/d58f8306-15c0-4bf9-8774-f3134cdd2629)

If you wish to play, check out the EditExec branch at https://github.com/drtonyr/wasp-os/

I've a few things to patch up, then I'll push.  I don't have a particular use for this code, it was just one of those weird things on my bucket list.

**Comment by fgaz** [comment]
Very nice! Do you think you could open a pr when it's complete?

**Reply by drtonyr** [reply]
Yes, that was the intention if someone else liked it.   I love the fact that I can show others that my watch runs Python.

There's quite a lot to do to make this 'complete'.   I went down the rabbit hole of drawing strings that fitted into bounding boxes and wiped the background without crashing - that led to three Pull Requests.   My level of git is such that it's easier to wait for those to be accepted then progress than work on two of my branches merged.

**Reply by fgaz** [reply]
Makes sense! Could you write down the numbers of the PRs you need so we can track them?

**Reply by drtonyr** [reply]
Sure.

#459 is really useful for drawing a full line of text (240 wide) and wiping what was there before.  At present I clear the whole screen and write again which doesn't look good.

#460 and #461 are related, at least one is needed to draw a 240 wide window starting from the left (I'm surprised that nobody else has found and fixed this).  #460 is for speed and will fix nearly everyone's problems, #461 gets to the real problem.

I'm working on a Tide Clock at the moment, it looks good but there is some hairly machine learning to finish off so that it really does give good time times.   I need to finish that off (which I will submit) before this editor.  The tide clock needs a data file, so hence issue #458

**Comment by Coldqueen19** [comment]
Did you push?

**Reply by drtonyr** [reply]
No, sorry.   The code is still available:  https://github.com/drtonyr/wasp-os/tree/EditExec
